include/aspect/introspection.h

```diff
@@ -294,6 +294,28 @@ namespace aspect
       IndexSets index_sets;
 
       /**
+       * A struct that describes the available methods for
+       * compositional fields. This type is used to determine how
+       * to advect a compositional field.
+       */
+      struct FieldMethod
+      {
+        enum kind
+        {
+          continuous_fem_field = 0,
+          discontinuous_fem_field = 1,
+          particles = 2
```

Comment:
Having read through the patch, the only place where you seem to make a distinction between the continuous and discontinuous case is in the place where you call the solver and, possibly, the limiter. But this is already superseded by #1109. Would it make sense to merge the first two enum entries and just call it `fem_field` or similar?